File: src/main/java/com/ngolajr/ispmedia/entities/Musica.java
package com.ngolajr.ispmedia.entities;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.UUID;

//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
@Entity
@ToString
@Getter
@Setter
public class Musica extends Conteudo{
    //@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @ManyToOne
    @JsonBackReference(value = "artista-musica")
    private Artista artista;
    @ManyToMany()
    private List<Artista> artists = new ArrayList<>();
    @ManyToOne()
    @JsonBackReference(value = "album-musica")
    private Album album;
    private int duration;
    private String path;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String letra;
    private Date dataLancamento;
    private int streams;
    public Musica(){
        super();
        this.streams=0;
    }
}

File: ISPMEDIA-ANGULAR/src/app/entities/Musica.ts
import { Conteudo } from './Conteudo';
import { Artista } from './Artista';
import { Album } from './Album';
import {Genero} from "./Genero";
import {User} from "./User";

export class Musica extends Conteudo {
  artists: Artista[];
  album: Album|null;
  duration: number;
  path: string;
  letra:string;
  dataLancamento:string;
  streams:number;
  artista:Artista=new Artista('', '', '', '', new Genero(), '', new User(), [], 0, 0);
  constructor(id: string, titulo: string, thumbNailUri: string, descricao: string,
              genero:Genero, editora:string, user:User, autor:Artista, artists: Artista[],
              album: Album|null, duration: number, path: string,
              letra:string, dataLancamento:string, streams:number) {
    super(id, titulo, thumbNailUri, descricao, genero, editora, user);
    this.artists = artists;
    this.album = album;
    this.duration = duration;
    this.path = path;
    this.letra = letra;
    this.dataLancamento= dataLancamento;
    this.streams = streams;
    this.artista=autor;
  }
}
